### PR TITLE
Asset management updates

### PR DESCRIPTION
**Pull request by alwaysblank**
Might add some more stuff to this, but I wanted to get a PR up to get 👀  on it.

If we want more here, just let me know. If there's some big stuff I'm missing, give me some pointers and I'll add it in.

### CODE REVIEW

**Review by retlehs — APPROVED** [review]

File: docs/acorn/2.x/assets-management.md [hunk 1]
```diff
@@ -1,5 +1,156 @@
...
+
+### asset()
+
+`asset(string $asset, ?string $manifest): Asset` - This returns an Asset object, which you can chain off of in several ways. `Roots\asset()` is the function to use if you want to deal with a single file, such as an image in your theme.
+  
+```php
+$name = 'sandwich.png';
+
+echo asset($name)->uri(); // https://tasty.food/theme/kitchen/assets/sandwich.png
+echo asset($name)->path(); // /srv/www/site/app/themes/kitchent/assets/sandwich.ong
```
Comment (swalkinshaw) [inline docs/acorn/2.x/assets-management.md]: typos for themes (theme), kitchent (kitchen), ong (png)

File: docs/acorn/2.x/assets-management.md [hunk 2]
```diff
@@ -1,5 +1,156 @@
...
+Acorn's asset management system is meant to make handling your assets easier and more fluent.
+It also integrates with the build process, automatically handling things like cache-busting filenames, manifests, chunking, etc.
+
+## Helper Functions
+
+Although you can interact directly with the [underlying systems](https://github.com/roots/acorn/tree/2.x/src/Roots/Acorn/Assets) if you like, Acorn makes some helper functions available to you in the `Roots` namespace:
+
+### asset()
+
+`asset(string $asset, ?string $manifest): Asset` - This returns an Asset object, which you can chain off of in several ways. `Roots\asset()` is the function to use if you want to deal with a single file, such as an image in your theme.
```
Comment (swalkinshaw) [inline docs/acorn/2.x/assets-management.md]: backtick `Asset`?

File: docs/acorn/2.x/assets-management.md [hunk 3]
```diff
@@ -1,5 +1,156 @@
...
+$name = 'sandwich.png';
+
+echo asset($name)->uri(); // https://tasty.food/theme/kitchen/assets/sandwich.png
+echo asset($name)->path(); // /srv/www/site/app/themes/kitchent/assets/sandwich.ong
+
+// When treated as a string, and Asset will return the URI of its asset:
+echo asset($name); // https://tasty.food/theme/kitchen/assets/sandwich.png
+```
+
+Asset has many methods:
```
Comment (swalkinshaw) [inline docs/acorn/2.x/assets-management.md:24]: backtick `Asset`?

File: docs/acorn/2.x/assets-management.md [hunk 4]
```diff
@@ -1,5 +1,156 @@
...
+- `uri(): string` - The public URI (which you may know colloquially as a URL) of the asset.
+- `path(): string` - The filesystem path of the asset on your server.
+- `exists(): bool` - Whether the asset exists.
+- `contents(): string` - The raw content of the asset. Useful for SVGs and the like; less useful for a PNG.
+- `base64(): string` - The raw content, but encoded in base64.
+- `dataUrl(): string` - The asset in data URL form, like you might use in CSS.
+- `dataUri(): string` - Alias of `dataUrl()`.
+- `contentType(): string|false` - The data type of the asset if it can be determined; `false` otherwise.
+- `mimeType(): string|false` - The MIME type of the asset, if it can be determined; `false` otherwise.
+- `file(): SplFileInfo` - An [SplFileInfo](https://www.php.net/manual/en/class.splfileinfo.php) instance of the asset.
```
Comment (swalkinshaw) [inline docs/acorn/2.x/assets-management.md]: backtick `SplFileInfo`?

File: docs/acorn/2.x/assets-management.md [hunk 5]
```diff
@@ -1,5 +1,156 @@
...
+  If you don't know what that is, you probably don't need it.
+
+### bundle()
+
+`bundle(string $bundle, ?string $manifest = null): Bundle` - This returns a Bundle object, which you can use to enqueue and interact with your JS and CSS dependencies.
```
Comment (swalkinshaw) [inline docs/acorn/2.x/assets-management.md:40]: backtick `Bundle`?

**Comment by retlehs** [comment]
pushed some updates based on scott's suggestions but did not do them all. please follow up if they're required - wanting to merge to get this up since we currently have nothing live

thank you @alwaysblank and thank you @swalkinshaw for reviewing

**Comment by alwaysblank** [comment]
@retlehs Thanks for handling that, sorry I haven't been more on the ball with this. It's been a hell of a month.

**Pull request merged**